_posts/2025-07-23-cockpit-343.md

```diff
@@ -0,0 +1,42 @@
+---
+title: Cockpit 343
+author: tmatus
+date: "2025-07-23"
+tags: cockpit, machines
+slug: cockpit-343
+category: release
+summary: "Improved VNC and consoles, various UX improvements."
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).
+We release regularly.
+
+Here are the release notes from Cockpit 343 and cockpit-machines 336:
+
+## Machines: Graphical VNC and serial consoles have been improved
+
+It is now easier to switch between these two console types, and also easier to launch a remote viewer. Cockpit can now add VNC and serial consoles to machines that are missing them, and can change the port and password of the VNC servers of individual machines.
+
+![Screenshot showing improved VNC view and includes new "Launch viewer" button.]({{ site.baseurl }}/images/343-machines-graphical-vnc-and-serial-console-improvemets.png)
+
+## Machines: Control VNC console resizing and scaling
+
+The VNC viewer allows changing its resizing and scaling behavior. You can now avoid any scaling and get a 1-to-1 pixel perfect view of a remote desktop, you can use local scaling or remote scaling.
+It is possible to avoid sending resize requests to a remote desktop.
```

Comment:
Proposal:

You can now explicitly control the graphical console's resizing and scaling behaviour in expanded mode:
 *  "No scaling or resizing" gets you an 1-to-1 pixel perfect view of a remote desktop, but possibly with scroll bars, depending on the browser window size.
 * "Local scaling" scales the console in the browser and thus is compatible with small browser windows and large remote resolutions, but is harder to read.
 * "Remote resizing" asks the remote machine to set a native resolution that matches Cockpit's console view. This keeps crisp graphics, but may not be compatible with all operating systems.